Odata-docs/webapi-8/function-routing.md

```diff
@@ -0,0 +1,365 @@
+GET http://localhost:5000/odata/Employees(1)/GetRating()
+```
+
+An alternative URL for invoking the same function is: `http://localhost:5000/odata/Employees(1)/Default.GetRating()`.
+
+For the above request to be conventionally-routed, a controller action named `GetRating` that accepts the key parameter is expected. The controller action should be decorated with a `HttpGet` attribute:
+```csharp
+[HttpGet]
+public ActionResult GetRating([FromRoute] int key)
+{
+    var employee = employees.SingleOrDefault(d => d.Id.Equals(key));
+
+    if (employee == null)
+    {
+        return NotFound();
+    }
+
+    return Ok(employee.PerfRating);
+}
+```
+
+The following JSON payload shows the expected response:
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Edm.Int32",
+    "value": 8
+}
+```
+
+### Invoking a function bound to a collection of derived entities or derived singleton
+The route template for this request is: `GET ~/{entityset}|{singleton}/{cast}/{function}`
+
+The following request invokes the `GetHighestBonus` function bound to the collection of `Manager` derived entities. The URL for the function is the function name appended to the collection of derived entities' URL:
+```http
+GET http://localhost:5000/odata/Employees/FunctionRouting.Models.Manager/GetHighestBonus()
+```
+
+An alternative URL for invoking the same function is: `http://localhost:5000/odata/Employees(1)/Default.GetHighestBonus()`.
+
+For the above request to be conventionally-routed, a controller action named `GetHighestBonusOnCollectionOfManager` that accepts no parameter is expected. The controller action should be decorated with a `HttpGet` attribute.
+```csharp
+[HttpGet]
+public ActionResult GetHighestBonusOnCollectionOfManager()
+{
+    var managers = employees.OfType<Manager>().ToArray();
+
+    if (managers.Length < 1)
+    {
+        return NoContent();
+    }
+
+    return Ok(managers.Select(d => d.Bonus).OrderByDescending(d => d).First());
+}
+```
+
+The following JSON payload shows the expected response:
+```json
+{
+    "@odata.context": "http://localhost:5000/odata/$metadata#Edm.Decimal",
+    "value": 3700
+}
+```
+
+### Invoking a function bound to a derived entity
+The route templates for this request are:
+- `POST ~/{entityset}({key})/{cast}/{function}`
+- `POST ~/{entityset}/{key}/{cast}/{function}`
+
+The following request invokes the `GetBonus` function bound to employee 5 (a manager). The URL for the function is the function name appended to the derived entity's URL:
+```http
+GET http://localhost:5000/odata/Employees(5)/FunctionRouting.Models.Manager/GetBonus()
+```
+
+An alternative URL for invoking the same function is: `http://localhost:5000/odata/Employees(5)/FunctionRouting.Models.Manager/Default.GetBonus()`.
+
+For the above request to be conventionally-routed, a controller action named `GetBonusOnManager` that accepts no parameter is expected. The controller action should be decorated with a `HttpGet` attribute.
```

Comment:
here you've written it accepts no parameter but has a key parameter?

Reply:
@ElizabethOkerio Good catch! Thanks